README.md

```diff
@@ -410,7 +423,60 @@ module.exports = {
 
 #### `minimizer`
 
-Type: `Object|Array<Object>`
+Type:
+
+```ts
+type minimizer =
+  | {
+      implementation: (
+        original: WorkerResult,
+        options?:
+          | {
+              [key: string]: any;
+            }
+          | undefined
+      ) => Promise<WorkerResult> & {
+        setup?: (() => void) | undefined;
+        teardown?: (() => void) | undefined;
+      };
+      options?:
+        | {
+            [key: string]: any;
+          }
+        | undefined;
+      filter?: (source: Buffer, sourcePath: string) => boolean | undefined;
+      filename?:
+        | string
+        | ((
+            pathData: {
+              filename?: string | undefined;
+            },
+            assetInfo?: import("webpack").AssetInfo | undefined
+          ) => string)
+        | undefined;
+    }
+  | Array<{
+      implementation: (
+        original: WorkerResult,
+        options?: InferDefaultType<T> | undefined
+      ) => Promise<WorkerResult> & {
+        setup?: (() => void) | undefined;
+        teardown?: (() => void) | undefined;
+      };
+      options?: InferDefaultType<T> | undefined;
+      filter?: (source: Buffer, sourcePath: string) => boolean | undefined;
+      filename?:
+        | string
+        | ((
+            pathData: {
+              filename?: string | undefined;
+            },
+            assetInfo?: import("webpack").AssetInfo | undefined
+          ) => string)
+        | undefined;
+    }>;
+```
```

Comment:
Let improve and should what is `WorkerResult` and simplify `InferDefaultType` using inline here (because it is internal)